pkg/apis/pipeline/v1beta1/container_conversion.go

```diff
@@ -0,0 +1,160 @@
+package v1beta1
+
+import (
+	"context"
+
+	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+)
+
+func (s Step) convertTo(ctx context.Context, sink *v1.Step) {
+	sink.Name = s.Name
+	sink.Image = s.Image
+	sink.Command = s.Command
+	sink.Args = s.Args
+	sink.WorkingDir = s.WorkingDir
+	sink.EnvFrom = s.EnvFrom
+	sink.Env = s.Env
+	sink.Resources = s.Resources
+	sink.VolumeMounts = s.VolumeMounts
+	sink.VolumeDevices = s.VolumeDevices
+	sink.ImagePullPolicy = s.ImagePullPolicy
+	sink.SecurityContext = s.SecurityContext
+	sink.Script = s.Script
+	sink.Timeout = s.Timeout
+
+	sink.Workspaces = nil
+	for _, w := range s.Workspaces {
+		new := v1.WorkspaceUsage{}
+		w.convertTo(ctx, &new)
+		sink.Workspaces = append(sink.Workspaces, new)
+	}
+	sink.OnError = s.OnError
+	sink.StdoutConfig = (*v1.StepOutputConfig)(s.StdoutConfig)
+	sink.StderrConfig = (*v1.StepOutputConfig)(s.StderrConfig)
+
+	// TODO(#4546): Handle deprecated fields
```

Comment:
no, since we don't have the v1 version added to our CRD yet. I have a follow-up PR almost ready that adds conversion for Resources. If we did have v1 versions in the task crd, conversion wouldn't fail, but it would silently drop any resources the user has defined. (I think this may only be backwards incompatible once we update the storage version of the crd but I'm not completely sure)